example/test/function_component_test.dart

```diff
@@ -183,6 +183,80 @@ UseRefTestComponent(Map props) {
   ]);
 }
 
+var FancyInput = react.forwardRef((props, ref) {
+  var inputRef = useRef();
+
+  useImperativeHandle(
+    ref,
+    () => ({
+      'focus': () {
+        inputRef.current.focus();
+      }
+    }),
+  );
+
+  return react.input({
+    'ref': inputRef,
+    'value': props['value'],
+    'onChange': (e) => props['update'](e.target.value),
+    'placeholder': props['placeholder'],
+    'style': {'borderColor': props['hasError'] ? 'crimson' : '#999'},
+  });
+});
+
+var useImperativeHandleTestFunctionComponent =
+    react.registerFunctionComponent(UseImperativeHandleTestComponent, displayName: 'useImperativeHandleTest');
+
+UseImperativeHandleTestComponent(Map props) {
+  var city = useState('');
```

Comment:
#nit this could be final
```suggestion
  final city = useState('');
```